File: CycleChessBot/game/cell.py

from game.color import Color
from game.piece import Piece
from game.position import Position

class Cell():
    def __init__(self, color: Color, position: Position, piece: Piece | None = None) -> None:
        self.color = color
        self.piece = piece
        self.position = position
    
    def set_piece(self, piece: Piece | None) -> None:
        self.piece = None
        if piece is not None:
            self.piece = Piece(piece.piece_type, piece.color)

File: CycleChesscpp/app/src/main/java/com/serebryakov/cyclechesscpp/application/model/game/gamefield/Cell.kt
package com.serebryakov.cyclechesscpp.application.model.game.gamefield

import com.serebryakov.cyclechesscpp.application.model.game.BasePiece
import com.serebryakov.cyclechesscpp.application.model.game.CellColor
import com.serebryakov.cyclechesscpp.application.model.game.PieceColor
import com.serebryakov.cyclechesscpp.application.model.game.PieceType

class Cell(val color: CellColor) {
    var piece: BasePiece? = null

    fun hasPiece() = (piece != null)

    fun hasPieceAnotherColor(color: PieceColor) =
        hasPiece() && (piece!!.color != color)

    fun hasPieceSameColor(color: PieceColor) =
        hasPiece() && (piece!!.color == color)

    fun hasPieceTypeColor(color: PieceColor, type: PieceType) =
        hasPieceSameColor(color) && (piece!!.type == type)

    fun getTypePiece(): PieceType? = piece?.type

    fun getAnotherColor() =
        if (color == CellColor.white) CellColor.black else CellColor.white
}
